### PR TITLE
Remove kinetic group additivity methods from code and documentation

### PR DESCRIPTION
**Pull request by jonwzheng**
### Motivation or Problem
See #2605. We have decided to remove group additivity as a valid estimator for kinetics. Now, only rate rules will be supported moving forward.
A previous commit (9f293bec90a093cf23b42089bb7bb0b14e05a2a7) removed the main group additivity estimator methods. However, there were still a few references to kinetic group additivity, including as the default method for `get_kinetics` if `kinetics_estimator` is not provided. There were also some mentions in the documentation and docstrings that suggest group additivity is supported.

### Description of Changes
This PR removes most mentions of kinetic group additivity from the code, documentation, and docstrings. 

However, this commit does _not_ remove the `KineticsGroup` class (see `rmgpy/data/kinetics/groups.py`), which is used extensively in the definition of a `KineticsFamily`. I would welcome some discussion on whether we should remove those and rewrite the kinetics family code as well, or if we want to keep it for compatibility / future reference/use etc.

### Testing
A local version of RMG-website using this commit hash seems to return the same kinetics as is currently returned on RMG-Py.

### CODE REVIEW

**Comment by JacksonBurns** [comment]
Re-running CI in the hope that the failures were spurious Julia dependency stuff.

**Comment by JacksonBurns** [comment]
Branch was outdated - rebased.

**Comment by JacksonBurns** [comment]
@jonwzheng I think there is just a simple formatting error in this table: https://github.com/ReactionMechanismGenerator/RMG-Py/actions/runs/7932965819/job/21660635035?pr=2612#step:9:335

**Comment by jonwzheng** [comment]
> @jonwzheng I think there is just a simple formatting error in this table: https://github.com/ReactionMechanismGenerator/RMG-Py/actions/runs/7932965819/job/21660635035?pr=2612#step:9:335

fixed this in the latest commit!

File: rmgpy/data/kinetics/family.py [hunk 1]
```diff
@@ -2396,23 +2396,20 @@ def get_reaction_template(self, reaction):
...
-        RMG-Java behavior, and default RMG-Py behavior).
+        `template` and reaction-path `degeneracy`. There is currently only one method to use:
+        'rate rules' (old RMG-Java behavior, and default RMG-Py behavior). Group additivity was removed in August 2023.
         
         Returns a tuple (kinetics, entry):
         If it's estimated via 'rate rules' and an exact match is found in the tree,
         then the entry is returned as the second element of the tuple.
-        But if an average is used, or the 'group additivity' method, then the tuple
-        returned is (kinetics, None).
+        But if an average is used,then the tuple returned is (kinetics, None).
```
Comment (JacksonBurns) [inline rmgpy/data/kinetics/family.py:2405]: Sorry I didn't catch this before - missing a space between the comma and `then`

**Review by JacksonBurns — APPROVED** [review]
Thanks for the patch! Enabling auto-merge

**Comment by JacksonBurns** [comment]
@jonwzheng could you check the live docs pages online and see if the changes here are reflected there?

**Comment by jonwzheng** [comment]
The live docs don't show this yet; see [this](https://reactionmechanismgenerator.github.io/RMG-Py/reference/data/templatereaction.html?highlight=rate%20rules) and [this](https://reactionmechanismgenerator.github.io/RMG-Py/users/rmg/kinetics.html?highlight=rate+rules).

**Comment by JacksonBurns** [comment]
Ugh, the latest doc building action fixes still aren't working. @jonwzheng could you put another pair of eyes on this workflow file and tell me if you see what is wrong? https://github.com/ReactionMechanismGenerator/RMG-Py/blob/main/.github/workflows/docs.yml

Here is the workflow run from merging this PR, it seems like the building worked but then the push failed?? https://github.com/ReactionMechanismGenerator/RMG-Py/actions/runs/7935329144/job/21668223608

**Comment by jonwzheng** [comment]
I'm looking into this and it looks like documentation build workflow has had this issue as early as last week: see [this for example](https://github.com/ReactionMechanismGenerator/RMG-Py/actions/runs/7876110961/job/21489442930).

Yeah, it looks like it's something specifically with the push step even though the build works. I'm not too familiar with how the workflow automation works, so I don't see anything that strikes me as particularly wrong.
The error `nothing to commit, working tree clean` is really weird. Is there any chance that the `/build/html` in the `Publish documentation` action is different from the `build/html/` compiled during `Make documentation - to publish`? I'm not sure how else to explain that there's nothing to commit unless we're operating in the original (rather than re-created) directory.

**Comment by rwest** [comment]
I think there could be a difference between `documentation/build/html` and `build/html`  ? 
Seem to be doing the git actions (checking out, committing, etc.) in the latter, but building the documentation in the former?
Could add some path printing stuff to debug, or could try something like https://github.com/mxschmitt/action-tmate to poke around and see what's happening.

**Pull request merged**